src/vscode-azureappservice.api.d.ts

```diff
@@ -0,0 +1,23 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+export interface AzureAppServiceExtensionApi {
+    apiVersion: string;
+    /**
+     * Creates a web app with the given options.  Empty properties will be their default value or the user will be prompted for it
+     *
+     * @param subscriptionId The subscription id of an Azure subscription: "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
+     * @param runtime The runtime must be formatted in the following: "NODE|10.14"
+     */
+    createWebApp(createOptions: {
+        subscriptionId?: string,
+        siteName?: string,
+        rgName?: string,
+        os?: WebsiteOS,
+        runtime?: string
```

Comment:
I still feel like this has to be more options than necessary. If they know all these values - they could pretty much create the site themselves. In that case, I would guide them to [SiteCreateStep](https://github.com/microsoft/vscode-azuretools/blob/master/appservice/src/createAppService/SiteCreateStep.ts) and they can basically just copy that single file instead of leveraging our extension.

If they _don't_ know all these values (which seems more likely), then we should remove them from the API so that we don't have to support it going forward.

Reply:
Well, they only want to specifically make Linux apps that are node|lts or python|3.7.  Should I just make the command more inflexible and add support as needed?  i.e. change it so that runtime is a type that accepts either node or python?

Reply:
Also the intern team wants support for deploy as well.

Reply:
Adding an API should be a very deliberative process and has a higher bar for merging vs. typical PRs because it deals with external teams and backwards compatibility. I don't think `inflexible` is the right word, but the API should definitely be as limited as you can make it.

Based on your comment, it seems like they only want to specify `os` and `runtime`, which means you should remove `siteName`, `rgName`, and `subscriptionId`. In addition - would our default runtime/OS recommendations work for them? Why do they need to specify specific versions?

Reply:
Well, they specifically requested to be able to make their own `siteName`, `rgName`, and to be able to pick the subscription that they want.  I think they want as little of user interactivity as possible.

I'll ask them if they plan on having the user or the dev team decide on the siteName and rgName, but subscription id does seem like the user would have to select.

As far as the runtimes, I think that for their specs, they are only supporting node/python projects, so our list of runtimes would confuse their users since we display everything even if the recommended runtimes are on top.

Reply:
https://us-prod.asyncgw.teams.microsoft.com/v1/objects/0-cus-d9-f4c1c20924fc926ed1a59e1c1fd58787/views/imgo

Since it's all wizard-based, our extension's quickpick steps wouldn't fit their needs.  It seems like it might be easier for the team to just use the azure-arm-resources API or copy our SiteCreateStep.ts from here https://github.com/microsoft/vscode-azuretools/blob/master/appservice/src/createAppService/SiteCreateStep.ts

Reply:
Can't see that image, but yeah it sounds like they shouldn't go through us if they don't want user interactive stuff